app/controllers/git_controller.rb

```diff
@@ -0,0 +1,37 @@
+require 'json'
+require 'git'
+require 'fileutils'
+
+class GitController < ApplicationController
+
+  def pull
+    conf = get_config
+    puts conf
+    if conf
+      clone_repo(conf['repo'], conf['branch'])
+    end
+    render :temp
+  end
+
+  def get_config
```

Comment:
`got_config` should be generalized, in an application level helper class, not in a controller. Additionally this should probably be in a totally different PR from a different branch as it is a separate feature, not actually related to, only required by the 'git' feature.